pkg/apis/sriovnetwork/v1/helper.go

```diff
@@ -80,19 +80,19 @@ func (p *SriovNetworkNodePolicy) Apply(state *SriovNetworkNodeState) {
 	state.Spec.Interfaces = append(state.Spec.Interfaces, interfaces...)
 }
 
-func (s *SriovNetworkNicSelector) Selected(iface *InterfaceExt) bool {
-	if s.Vendor != "" && s.Vendor != iface.Vendor {
+func (s *SriovNetworkNodePolicySpec) Selected(iface *InterfaceExt) bool {
+	if s.NicSelector.Vendor != "" && s.NicSelector.Vendor != iface.Vendor {
 		return false
 	}
-	if s.DeviceID != "" {
-		if ((iface.NumVfs == 0 && s.DeviceID != iface.DeviceID) || (iface.NumVfs > 0 && s.DeviceID != SriovPfVfMap[iface.DeviceID])) {
+	if s.NicSelector.DeviceID != "" {
+		if ((s.NumVfs == 0 && s.NicSelector.DeviceID != iface.DeviceID) || (s.NumVfs > 0 && s.NicSelector.DeviceID != SriovPfVfMap[iface.DeviceID])) {
```

Comment:
If `s.NicSelector.DeviceID` is a VF ID and `iface.DeviceID` is equal to this VF ID, it seems that `iface` will be listed in the config spec, but it's not configurable because it's a VF device.

Reply:
The for loop in Apply() only walk through all the PFs. And in spec of NodeState, the desired configuration is indexed by PF only.

Reply:
ok, thanks for the explanation, I think it should be good to go now.